main.py

```diff
@@ -190,23 +223,13 @@ def evaluate_genomes(genomes, config):
     food_list = get_food(NUM_FOOD)
 
     while True:
-        if time.time() >= end_round_time or len(players_list) < 2:
-            for player_index, player in enumerate(players_list):
-                genomes_list[player_index].fitness = calculate_fitness(player)
-            print("Round time done")
-            for i in reversed(range(3)):
-                print(f"Starting next generation in {i+1}...", end="\r")
-                time.sleep(1)
+        if calculate_all_fitness(end_round_time, players_list, genomes_list):
             break
 
         # Continue the game loop
         players_list = check_for_game_events(players_list)
 
-        if len(players_list) == 0:
-            run = False
-            break
-
-        if goal_reached(players_list):
+        if len(players_list) == 0 or goal_reached(players_list):

```

Comment:
I argue that these evaluations could be moved to the above if statement too as we'll just be ending the game anyway